ETTrace/ETTrace/EMGPerfAnalysis.mm

```diff
@@ -190,19 +287,47 @@ + (NSString *)deviceName {
 + (void)stopRecording {
     sStacksLock.lock();
     NSMutableArray <NSDictionary <NSString *, id> *> *stacks = [NSMutableArray array];
-    for (const auto &cStack : *sStacks) {
-        NSMutableArray <NSNumber *> *stack = [NSMutableArray array];
-        // Add the addrs in reverse order so that they start with the lowest frame, e.g. `start`
-        for (int j = (int)cStack.frameCount - 1; j >= 0; j--) {
-            [stack addObject:@((NSUInteger)cStack.frames[j])];
+    NSMutableDictionary <NSString *, NSDictionary<NSString *, id> *> *threads = [NSMutableDictionary dictionary];
+    if (sRecordAllThreads) {
+        std::map<unsigned int, Thread *>::iterator it;
+        for (it = sThreadsMap->begin(); it != sThreadsMap->end(); it++) {
+            NSMutableArray <NSDictionary <NSString *, id> *> *threadStacks = [NSMutableArray array];
+            Thread thread = *it->second;
+            for (const auto &cStack : *thread.stacks) {
+                NSMutableArray <NSNumber *> *stack = [NSMutableArray array];
+                // Add the addrs in reverse order so that they start with the lowest frame, e.g. `start`
+                for (int j = (int)cStack.frameCount - 1; j >= 0; j--) {
+                    [stack addObject:@((NSUInteger)cStack.frames[j])];
+                }
+                NSDictionary *stackDictionary = @{
+                    @"stack": [stack copy],
+                    @"time": @(cStack.time)
+                };
+                [threadStacks addObject:stackDictionary];
+            }
+            NSString *threadId = [[NSNumber numberWithUnsignedInt:it->first] stringValue];
+            threads[threadId] = @{
+                @"name": [NSString stringWithFormat:@"%s", thread.name],
+                @"stacks": threadStacks
+            };
         }
-        NSDictionary *stackDictionary = @{
-            @"stack": [stack copy],
-            @"time": @(cStack.time)
-        };
-        [stacks addObject:stackDictionary];
+        sThreadsMap->empty();
+    } else {
+        for (const auto &cStack : *sStacks) {
```

Comment:
It would be nice to pull this into a small function like `-(NSArray< NSNumber *> *)arrayFromCStack:(std::Vector<Stack> *)cStack`